feat: setup the User API service \n

asiimwearnold opened the pull request:
This feature will control the authentication and authorization for the application.\n

Closes #6

asiimwearnold committed:
feat: setup the User API service \n

This feature will control the authentication and authorisation for the application.\n

Fixes #6

amkayondo commented on src/services/user_api/.env:
```diff
@@ -0,0 +1 @@
+DB_CONNECT = mongodb+srv://user1:tKFnEm8vhLI5txYw@farmrail-qj0hk.gcp.mongodb.net/test?retryWrites=true&w=majority
```
Don't include the .env file, it should be private and also all the .env variables are set on where the code runs like on Heroku, testing environments or the server.  So add it to the git ignore file.

kmwamasali replied:
export database connection logic to a separate file and export as constructor object. Clearer documentation this will be provided

amkayondo commented on src/services/user_api/index.js:
```diff
@@ -0,0 +1,17 @@
+const app = express();
+const dotenv = require('dotenv');
+const mongoose = require('mongoose');
+// import Routes
```
Comments are unnecessary as the import naming convention is enough to show what the import does like **import AppRoute from "./routes';**

amkayondo commented on src/services/user_api/index.js:
```diff
@@ -0,0 +1,17 @@
+const dotenv = require('dotenv');
+const mongoose = require('mongoose');
+// import Routes
+const authRoute = require('./routes/auth');
```
Migrate to using es6 import feature so as to meet the es6 code standard, learn more from here [ES6 IMPORT](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)

amkayondo commented on src/services/user_api/routes/auth.js:
```diff
@@ -0,0 +1,21 @@
+const router = require('express').Router();
+const User = require('../models/user');
+
+// routes "middle-ware"
```
remove the unnecessary inline comments

asiimwearnold replied:
Done

asiimwearnold replied:
> remove the unnecessary inline comments

Done

amkayondo commented on src/services/user_api/routes/auth.js:
```diff
@@ -0,0 +1,21 @@
+
+//router.post('/login');
+
+module.exports = router;
```
migrate to the es6 export by default, learn more [here](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export)

kmwamasali replied:
Incldude unit tests for this functions.

asiimwearnold replied:
> migrate to the es6 export by default, learn more [here](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export)

Done

asiimwearnold replied:
> Incldude unit tests for this functions.

In progress...

amkayondo commented on src/services/user_api/package.json:
```diff
@@ -0,0 +1,21 @@
+    "express": "^4.17.1",
+    "mongoose": "^5.9.12"
+  },
+  "devDependencies": {
```
add @babel to your dependencies so as you can easily use ES6 functionality in your code. Also, learn more about **@babel build, @babel/node, @babel/preset-env**

asiimwearnold replied:
> add @babel to your dependencies so as you can easily use ES6 functionality in your code. Also, learn more about **@babel build, @babel/node, @babel/preset-env**

Duly Noted 👍  added!

amkayondo commented on src/services/user_api/models/user.js:
```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
```
migrate to es6 import functionality

amkayondo commented on src/services/user_api/models/user.js:
```diff
@@ -0,0 +1,28 @@
+  }
+});
+
+module.exports = mongoose.model('User', userSchema);
```
migrate to export by default, also take time to know when to use **module.exports, export by default and export**

amkayondo reviewed (changes requested):
Fix the required changes highlighted in your commit

kmwamasali commented on src/services/user_api/index.js:
```diff
@@ -0,0 +1,17 @@
+// route "middle-ware"
+app.use('/api/user', authRoute);
+
+app.listen(3000, () => console.log('server is Up and running'));
```
create unit test for main server

kmwamasali commented on src/services/user_api/package.json:
```diff
@@ -0,0 +1,21 @@
+  "description": "Api that allows users to login, create acc, delete account,authenticate account",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
setup scripts for testing and linting

asiimwearnold committed:
RF: improving code of the ft-user-api

Removing the .env file

asiimwearnold committed:
improving code of the ft-user-api

Removing the .env file

asiimwearnold committed:
RF:improving code of the ft-user-api

Removing the .env
>> >> Migrating to the es6 imports
>> >> Database connection logic to a sepperate file
>> >> Removidng commnets
>> >> Creating unit tests for the main server[Not yet complete test soon to be added]

Fixes #06

amkayondo commented on src/services/user_api/src/index.js:
```diff
@@ -15,4 +15,4 @@ mongoose
   .then(() => console.log('connected to db'));
 
-app.listen(3000, () => console.log(`server is Up and running on port:${config.port}`));
+app.listen(3000, () => console.log(`server is Up and running on port:${3000}`));
```
store the env variables like **the port** in the env file and access it from there

amkayondo commented on src/services/user_api/default.txt:
```diff
@@ -0,0 +1,10 @@
+RF: improving code of the ft-user-api
```
remove this default.txt file as all the messages should be in your commit message

amkayondo reviewed (changes requested):
remove the default.txt file and make also the needed changes in your code

kmwamasali reviewed (approved):
Most tests are in check with a few pending, I will be merged

kmwamasali reviewed (approved):
linting errors still present otherwise good task and will add the necessary requests in refactor
